### PR TITLE
JSON dump

### PR DESCRIPTION
**Pull request by jaimeiniesta**
This is a WIP for the JSON dump option, discussed in #15

### CODE REVIEW

**Comment by WilcoFiers** [comment]
Hi Jaime. Looking good so far, but I would suggest we use `--reporter`, and allow users to set which reporter they want to use for their JSON object. That way we hit two birds with one stone. I didn't implement that option initially, because for some of the reporters the CLI output wouldn't work anymore (as the format changed). But if we're outputting raw JSON, we might as well allow users to define which reporter they want. That way you could for instance have it only output violations.

**Comment by jaimeiniesta** [comment]
For that I'll need to investigate more. I see in https://github.com/dequelabs/axe-core/blob/develop/doc/API.md that the `reporter` option can take `v1` or `v2`.

But then in https://github.com/dequelabs/axe-core/tree/develop/lib/core/reporters I see there are more reporters. Should all these be an option as well?

**Comment by WilcoFiers** [comment]
If we just pipe the value that the user puts in --reporter into the reporter option, it'll work. I'll check with the team if we want to document the other reporters. I can imagine particularly for CLI it would be useful to use the no-passess reporter.

**Comment by jaimeiniesta** [comment]
So you mean to use both options at the same time, like `--json-dump --reporter=no-passess`, right?

File: index.js [hunk 1]
```diff
@@ -64,7 +64,7 @@ axeTestUrls(urls, program, {
 	 */
 	onTestComplete: function logResults(results) {
     if (program.jsonDump) {
-      console.log(results)
+      console.log(JSON.stringify(results))
```
Comment (jaimeiniesta) [inline index.js]: @WilcoFiers I've run into a weird bug here. For most pages this works fine, if you check out my fork you can get a JSON output like this:

```
./axe-cli -j http://example.com
```
But for some pages I get a `Converting circular structure to JSON` error when trying to convert the results to JSON, for example:

```
$ ./axe-cli -j http://validationhell.com
Error: "Converting circular structure to JSON"
 $s TypeError: Converting circular structure to JSON
    at JSON.stringify (<anonymous>)
    at Object.logResults [as onTestComplete] (/Users/jaime/code/axe-cli/index.js:68:24)
    at /Users/jaime/code/axe-cli/lib/axe-test-urls.js:66:13
    at ManagedPromise.invokeCallback_ (/Users/jaime/code/axe-cli/node_modules/selenium-webdriver/lib/promise.js:1384:14)
    at TaskQueue.execute_ (/Users/jaime/code/axe-cli/node_modules/selenium-webdriver/lib/promise.js:3092:14)
    at TaskQueue.executeNext_ (/Users/jaime/code/axe-cli/node_modules/selenium-webdriver/lib/promise.js:3075:27)
    at asyncRun (/Users/jaime/code/axe-cli/node_modules/selenium-webdriver/lib/promise.js:2935:27)
    at /Users/jaime/code/axe-cli/node_modules/selenium-webdriver/lib/promise.js:676:7
    at process._tickCallback (internal/process/next_tick.js:103:7)
^C
```

The script hangs there and I have to kill it with control+C.

Does it ring a bell? Is there a bug in the results structure? Should we use something like [json-circular](https://www.npmjs.com/package/json-circular) instead?

**Comment by jaimeiniesta** [comment]
I've refactored this, found the good spot where to convert the results to JSON.

Still, don't like it as there are a lot of console logs that need to be silenced in this case. I think we should split this into separate reporters.

**Comment by jaimeiniesta** [comment]
I'm finding this is going to be more complicated than just dealing with the saved file, so I'm not going to continue with this PR at least in the short term.

Thanks for considering it!

**Pull request closed without merging**